### DISCUSSION TITLE
Unable to create PR due to "Validation failed: must be a collaborator"

### DISCUSSION THREAD

**Discussion by jtbandes** [community/community#61472] [Pull Requests]
### Select Topic Area

Bug

### Body

(More context: https://github.com/qchateau/conan-center-bot/issues/111)

There is a bot ([conan-center-bot](https://github.com/qchateau/conan-center-bot)) that pushes branches to [a fork](https://github.com/qchateau/conan-center-index) of [conan-center/conan-center-index](https://github.com/conan-io/conan-center-index).

As a third party (neither the owner of the upstream repo nor the owner of the fork) I can visit [this page](https://qchateau.github.io/conan-center-bot/#/updatable), and click a link to open a PR for one of these branches. For example: https://github.com/qchateau/conan-center-index/pull/new/ccb-mcap-1.2.0

This redirects me to the "compare" page at https://github.com/conan-io/conan-center-index/compare/master...qchateau:conan-center-index:ccb-mcap-1.2.0?expand=1

<img width="1113" alt="image" src="https://github.com/community/community/assets/14237/e6ee2368-0214-43e1-8ab0-9de64b2f31eb">

**However, when I click "Create pull request", I get this error:** 

<img width="494" alt="image" src="https://github.com/community/community/assets/14237/e4a20ebe-893b-42fe-93fb-c276205f3cef">

What's going on here? Is this a new requirement? Is it controlled by some repo settings, either on the original upstream repo or on the fork? The error message doesn't provide enough information to find any more info in the docs.

**Comment by Zudjo** [comment]
I have this problem too. Did you manage to solve it?

**Reply by jtbandes** [reply]
Nope!

**Comment by a4z** [comment]
same situation here, please see 
https://github.com/cross-language-cpp/djinni-generator/discussions/154

any idea what we could do?

**Comment by Zudjo** [comment]
I managed to "resolve" this.

I'm not sure if this was the case for you too, but I got this error because i was trying to make a pull request directly from github, without passing from the cloned repo.

**Comment by a4z** [comment]
Thanks for the info!
I can not say what our problem was, since it was not my problem, but from the person who wanted to create a PR to out repo.

**Comment by vient** [comment]
Stumbled into same problem - ended up making my fork of base repository, opening an outside commit there, creating new branch from it (enter new branch name, github will ask if you want to create a new branch from this commit), and finally creating pull request from this new branch.

Guess the easier way would be to fork derived repository where needed branch already exists, and then create a pull request from it.

**Comment by Neustradamus** [comment]
This problem always exists, I have the same error.

**Comment by mabartos** [comment]
We're facing the same issue. I'd suspect the issue might be that the fork belongs to the organization itself and not a real user. When I have a commit created by our GH App and push it to my personal fork and then compare it with my personal fork, I can create the PR. 

It'd be great if would be possible to compare changes with the repository belonging to an organization and then create the PR.

But it's not verified, and it's only a guess at this moment. I'll try to verify it.

EDIT: However, I see that you're trying to compare it with fork belonging to user @qchateau and not any organization, so it might be some wider problem.

**Comment by erkinalp** [comment]
This is a serious problem. If we aren't allowed to suggest another public fork's published code for a PR, why are we sharing our code in the first place?

**Reply by erkinalp** [reply]
Reiterating again. Not every open source software is developed in a top-down dispatch-and-assign-tasks fashion, and ability of a contributor to suggest a change from another for a PR is a very important way to draw attention to improvements and fixes from fringe forks.

**Reply by erkinalp** [reply]
Before saying it's controversial, it's allowed on forgejo albeit with a roundabout UX: https://codeberg.org/forgejo/forgejo/issues/9449

**Comment by omprxkash** [comment]
Select Topic Area:
Bug

Body:

(More context: [qchateau/conan-center-bot#111](https://github.com/qchateau/conan-center-bot/issues/111)
)

There is a bot (conan-center-bot) that pushes branches to a fork of conan-center/conan-center-index.

As a third party (i.e., I’m neither the owner of the upstream repo nor the owner of the fork), I can visit this page and click a link to open a PR for one of these branches.
For example:
👉 https://github.com/qchateau/conan-center-index/pull/new/ccb-mcap-1.2.0

This redirects me to the compare page:
https://github.com/conan-io/conan-center-index/compare/master...qchateau:conan-center-index:ccb-mcap-1.2.0?expand=1

📸 Screenshot:

**Reply by erkinalp** [reply]
Yes you can open the page but the server won't authorise the PR. It'll fail with "Validation failed: must be a collaborator"
